packages/m2m-gateway/test/integration/eservices/suspendDescriptor.test.ts

```diff
@@ -0,0 +1,127 @@
+  const mockGetEservice = vi.fn(mockPollingResponse(mockApiEservice, 2));
+
+  mockInteropBeClients.catalogProcessClient = {
+    getEServiceById: mockGetEservice,
+    suspendDescriptor: mockSuspendDescriptor,
+  } as unknown as PagoPAInteropBeClients["catalogProcessClient"];
+
+  beforeEach(() => {
+    mockSuspendDescriptor.mockClear();
+    mockGetEservice.mockClear();
+  });
+
+  it("Should succeed and perform API clients calls", async () => {
+    const result = await eserviceService.suspendDescriptor(
+      unsafeBrandId(mockApiEservice.data.id),
+      unsafeBrandId(mockApiDescriptor.id),
+      getMockM2MAdminAppContext()
+    );
+
+    expect(result).toEqual(mockM2MEserviceDescriptorResponse);
+    expectApiClientPostToHaveBeenCalledWith({
+      mockPost: mockInteropBeClients.catalogProcessClient.suspendDescriptor,
+      params: {
+        eServiceId: mockApiEservice.data.id,
+        descriptorId: mockApiDescriptor.id,
+      },
+    });
+    expectApiClientGetToHaveBeenCalledWith({
+      mockGet: mockInteropBeClients.catalogProcessClient.getEServiceById,
+      params: { eServiceId: mockApiEservice.data.id },
+    });
+    expect(
+      mockInteropBeClients.catalogProcessClient.getEServiceById
+    ).toHaveBeenCalledTimes(2);
+  });
+
+  it("Should throw missingMetadata in case the eservice returned by the suspend call has no metadata", async () => {
+    mockSuspendDescriptor.mockResolvedValueOnce({
+      metadata: undefined,
+    });
+
+    await expect(
+      eserviceService.suspendDescriptor(
+        unsafeBrandId(mockApiEservice.data.id),
+        unsafeBrandId(mockApiDescriptor.id),
+        getMockM2MAdminAppContext()
+      )
+    ).rejects.toThrowError(missingMetadata());
+  });
+
+  it("Should throw missingMetadata in case the eservice returned by the polling GET call has no metadata", async () => {
+    mockGetEservice.mockResolvedValueOnce({
+      data: mockApiEservice.data,
+      metadata: undefined,
+    });
+
+    await expect(
+      eserviceService.suspendDescriptor(
+        unsafeBrandId(mockApiEservice.data.id),
+        unsafeBrandId(mockApiDescriptor.id),
+        getMockM2MAdminAppContext()
+      )
+    ).rejects.toThrowError(missingMetadata());
+  });
+
+  it("Should throw pollingMaxRetriesExceeded in case of polling max attempts", async () => {
+    mockGetEservice.mockImplementation(
+      mockPollingResponse(mockApiEservice, config.defaultPollingMaxRetries + 1)
+    );
+
+    await expect(
+      eserviceService.suspendDescriptor(
+        unsafeBrandId(mockApiEservice.data.id),
+        unsafeBrandId(mockApiDescriptor.id),
+        getMockM2MAdminAppContext()
+      )
+    ).rejects.toThrowError(
+      pollingMaxRetriesExceeded(
+        config.defaultPollingMaxRetries,
+        config.defaultPollingRetryDelay
+      )
+    );
+    expect(mockGetEservice).toHaveBeenCalledTimes(
+      config.defaultPollingMaxRetries
+    );
+  });
+});
```

Comment:
Could it make sense to add a test for the case where it throws `eserviceDescriptorNotFound`?
Since it's logic implemented in the m2m, I would add a test for that - we have similar tests for purpose versions logic (getPurposeVersion, createPurposeVersion, etc.)